### PR TITLE
Atmosphere compositions

### PR DESCRIPTION
**Pull request by Seggan**
## Changes
Added atmosphere compositions and made crops grow faster in high CO<sub>2</sub> concentrations.

## Checklist

- [x] I have fully tested the proposed changes and promise that they will not break world generation, mob spawning, and the like.
- [x] I followed the existing code standards and didn't mess up the formatting.
- [x] I did my best to add documentation to any public classes or methods I added.
- [x] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values

### CODE REVIEW

File: src/main/java/io/github/addoncommunity/galactifun/api/universe/attributes/atmosphere/Atmosphere.java [hunk 1]
```diff
@@ -32,18 +36,21 @@
...
+    @Getter
+    private final Map<AtmosphericComponent, BigDecimal> composition = new EnumMap<>(AtmosphericComponent.class);
     
     // builder's constructor
-    Atmosphere(double oxygenPercentage, double carbonDioxidePercentage, boolean weatherCycle, boolean storming, boolean thundering,
-               boolean flammable, @Nonnull World.Environment environment, @Nonnull AtmosphericEffect[] effects) {
-        
-        this.oxygenPercentage = oxygenPercentage;
+    Atmosphere(boolean weatherCycle, boolean storming, boolean thundering, boolean flammable,
+               @Nonnull World.Environment environment, Map<AtmosphericComponent, BigDecimal> composition,
```
Comment (Mooy1) [inline src/main/java/io/github/addoncommunity/galactifun/api/universe/attributes/atmosphere/Atmosphere.java:45]: no need for big decimal, float would work fine

File: src/main/java/io/github/addoncommunity/galactifun/api/universe/attributes/atmosphere/Atmosphere.java [hunk 2]
```diff
@@ -15,13 +18,14 @@
  */
 public final class Atmosphere {
 
-    public static final Atmosphere EARTH_LIKE = new AtmosphereBuilder().addOxygen(21).addCarbonDioxide(.004).enableWeather().build();
+    public static final Atmosphere EARTH_LIKE = new AtmosphereBuilder().enableWeather()
+        .addComponent(AtmosphericComponent.NITROGEN, 77.084) // subtracted 1 to allow water to fit in
+        .addComponent(AtmosphericComponent.OXYGEN, 20.946)
```
Comment (Mooy1) [inline src/main/java/io/github/addoncommunity/galactifun/api/universe/attributes/atmosphere/Atmosphere.java:23]: maybe add invividual methods for common ones, such as an addOxygen

**Review by Mooy1 — APPROVED** [review]
other than that lgtm

File: src/main/java/io/github/addoncommunity/galactifun/api/universe/attributes/atmosphere/Atmosphere.java (hunk 1 above)
Comment (Seggan) [reply src/main/java/io/github/addoncommunity/galactifun/api/universe/attributes/atmosphere/Atmosphere.java:45]: i need the accuracy when working with small impurities

**Pull request merged**